onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -20,27 +20,50 @@ class AttentionCPUBase : public AttentionBase {
 
   template <typename T>
   Status ApplyAttention(const T* Q,                           // Q data with shape BxNxSxH
-                        const T* K,                           // K data with shape BxNxSxH
-                        const T* V,                           // V value with size BxNxSxH_v
+                        const T* K,                           // K data with shape BxNxLxH
+                        const T* V,                           // V value with size BxNxLxH_v
                         const Tensor* mask_index,             // mask index. nullptr if no mask or its size is B
                         const Tensor* past,                   // past state
                         Tensor* output,                       // output tensor
+                        Tensor* present_k,                    // present K output tensor (if separating present KV)
+                        Tensor* present_v,                    // present V output tensor (if separating present KV)
                         int batch_size,                       // batch size (B)
-                        int sequence_length,                  // sequence length (S)
+                        int sequence_length,                  // sequence length of Q (S)
+                        int kv_sequence_length,               // sequence length of K or V (L)
                         int qk_head_size,                     // head size of Q or K (H)
                         int v_head_size,                      // head size of V (H_v)
                         int v_hidden_size,                    // hidden size of V (D_v)
                         const Tensor* relative_position_bias, // bias addition in QK. Its size is BxNxSxT
+                        bool separate_present_kv,             // whether to separate present KV into present K and present V
```

Comment:
Can this value be deduct from check whether the present_key/value is nullptr?

Reply:
Fixed